src/components/container/BloodPressureVisualization/BloodPressureVisualization.tsx

```diff
@@ -0,0 +1,257 @@
+        }
+
+        return MyDataHelps.querySurveyAnswers(queryParameters);
+    }
+
+    function initializeWeekPager() : Date {
+        const today = startOfDay(new Date());
+        const monday : Date = isMonday(today) ? today : previousMonday(today);
+        const sunday : Date = nextSunday(monday);
+        return monday;
+    }
+
+    function pageWeek(newStart: Date) {
+        pageWeeklyData( newStart );
+    }
+
+    function pageWeeklyData(startOfWeek: Date, freshData? : IBloodPressureDumbbell[]){
+        const useData = freshData ?? bpData;
+        const weekData : Dumbbell[] = [];
+        for (let i = 0; i < 7; i++) {
+            var currentDate = startOfDay(addDays(startOfWeek, i));
+            var dataForDay = useData.find( db => isEqual(db.date, currentDate));
+            if (!dataForDay)
+            {
+                var formattedDate = format(currentDate, "MM/dd");
+                weekData.push({xValue : formattedDate});
+            } 
+            else 
+            {
+                weekData.push(dataForDay.dumbbell);
+            }
+        }
+
+        setWeeklyDataForViz(weekData);
+        setStartOfWeek(startOfWeek);
+    }
+
+    function buildInterval(entries : number[]){
+        const min = Math.min(...entries);
+        const closedInterval : ClosedInterval = {values : [min]};
+        
+        var max = min;
+        if (entries.length > 1){
+            max = Math.max(...entries);
+        }
+        closedInterval.values.push(max);
+        return closedInterval;
+    }
+
+    function assignClass(avgSystolic : number, avgDiastolic : number){
+
+        if ((avgSystolic >=90 && avgSystolic <= 119) && (avgDiastolic >= 60 && avgDiastolic <= 79)){
+            return DumbbellClass.inRange;
+        }
+
+        return DumbbellClass.outOfRange;
+    }
+
+    useEffect(() => {
+        let debouncedInitialize = debounce(initialize, 500);
+ 
+        debouncedInitialize();
+        mdh.on("applicationDidBecomeVisible", debouncedInitialize);
+         
+        return () => {
+            mdh.off("applicationDidBecomeVisible", debouncedInitialize);
+        }
+     }, []);
+
+    if (loadingData) 
+    {
+        return <LoadingIndicator/>;
+    }
+    else
+    {
+        return (
+            <div className="bloodPressureVisualization">
+                <div className="bpTitle">Blood Pressure</div>
+                <div>
+                    <div className="pagerRow">
+                        <DateRangeNavigator intervalType="Week" intervalStart={startOfWeek} onIntervalChange={pageWeek}></DateRangeNavigator>
+                    </div>
+                    <DumbbellChart dumbbells={weeklyDataForViz} axis={axis} ></DumbbellChart>
+                </div>
+                <div className="buffer"></div>
+                <div>
+                    <Button onClick={() => mdh.startSurvey(props.surveyName ?? "")}><span className="buttonCaption">Log Blood Pressure</span><img src={plus}/></Button>
```

Comment:
The bottom left border radius is different from the rest of the button curves and I'm not sure why. 
1) I question if the button should be part of this component
2) Is debouce needed here? I understood that it would get rid of blinking, but I still see blinking.

Reply:
Why do you need a specific "plus" icon here?  Does the font awesome plus icon not work?  Not a big deal either way but seems unnecessary to bring in another svg